packages/gatsby-cli/src/create-cli.js

```diff
@@ -13,13 +15,15 @@ const handlerP = fn => (...args) => {
 }
 
 function buildLocalCommands(cli, isLocalSite) {
-  const defaultHost = `localhost`
+  const defaultHost = `0.0.0.0`
```

Comment:
Turns out Windows doesn't need us to listen on localhost, just that I guess the browser can't use 0.0.0.0 or something like that.